server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1695,8 +1697,30 @@ public AliasFilter buildAliasFilter(ClusterState state, String index, Set<String
     /**
      * Returns a new {@link QueryRewriteContext} with the given {@code now} provider
      */
-    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis) {
-        return new QueryRewriteContext(parserConfig, client, nowInMillis);
+    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis, IndicesRequest indicesRequest) {
+        // Skip unavailable indices
+        // TODO: Support metadata lookups for indices on remote clusters
+        IndicesOptions.Builder indicesOptionsBuilder = IndicesOptions.builder(indicesRequest.indicesOptions())
```

Comment:
Do we need to change the indices options? Shouldn't that be something the user does, or fail if indices options do not allow concrete indices to be resolved?

Reply:
Throwing an exception on an unavailable index here is effectively a breaking change. Previously, this method didn't check the validity of the indices named in the request, leaving that validation check further downstream. Throwing on an unavailable index here changes the failure mode for _all_ queries, which breaks CI tests.

Adjusting the indices options to skip unavailable indices also makes sense from the perspective that index metadata lookup is inherently an internal operation, and thus we can set the rules for how we want to do it. One common use case we need to handle is invalid user input, such as when the user provides an invalid index name. As stated above, we can't throw on this without creating a breaking change, but we can skip such index names and let the request fail further downstream as it would have done previously.

Of course, we should add tests for all this to validate :)

Reply:
Got it - thanks for clarifying. As long as this is done for the inference operation we should be safe 👍